### PR TITLE
Added a few SQL modes

### PR DESCRIPTION
**Pull request by atsage**
Fixes SQL injection.

Added SQL Mode and SQL Statement entry to the config.  SQL Mode allows selection of how the SQL statement is provided to the node.  msg.topic operates as the node currently operates.  Normal uses the SQL statement entered on the config.  Prepared uses the statement entered on the config and passes in parameters from msg.params.  SQL Statement is a ACE editor for SQL Queries.

### CODE REVIEW

**Comment by knolleary** [comment]
Hi, thanks for raising a pull request.

If you check out our [contribution guidelines](https://github.com/node-red/node-red/blob/master/CONTRIBUTING.md) you'll see we prefer feature request/suggestions are first discussed on the [mailing list](https://groups.google.com/forum/#!forum/node-red) or [slack team](https://nodered.org/slack). Note we also reserve the right to close PRs that have failed to follow our contribution guidelines.

This PR does not include any information about what it is adding. The one line title is not enough for us to understand what it is doing.

Please describe in detail what functionality this adds, what problems it solves etc.

Thanks!

**Comment by atsage** [comment]
Any updates on this PR?

**Comment by hardillb** [comment]
@atsage the edited description is still not really enough, you need to explain what problems it is solving. Also as noted these changes should have been discussed on the mailing list or slack first.

Also before any PR can be accepted you MUST sign the CLA.

File: storage/sqlite/sqlite.js [hunk 1]
```diff
@@ -33,29 +33,82 @@ module.exports = function(RED) {
...
             node.on("input", function(msg) {
-                if (typeof msg.topic === 'string') {
-                    //console.log("query:",msg.topic);
-                    var bind = Array.isArray(msg.payload) ? msg.payload : [];
-                    node.mydbConfig.db.all(msg.topic, bind, function(err, row) {
-                        if (err) { node.error(err,msg); }
-                        else {
-                            msg.payload = row;
-                            node.send(msg);
+                if (this.sqltype == "msg.topic"){
```
Comment (dceejay) [inline storage/sqlite/sqlite.js]: this will break all previous version of th node unless defaulted correctly.

**Comment by atsage** [comment]
Hey guys, finally came back around to finish this up.  I defaulted to msg.topic if nothing is set to keep backward compatibility.  Not sure what else to add to the description of what this PR does, the first post in the thread covers everything.  It's an enhancement to prevent SQL injection by providing prepared statements.

**Pull request merged**